### PR TITLE
W-11771438: [Spike] OAS serialization for parameters

### PR DESCRIPTION
**Pull request by juanchib**
## Description

This draft shows a possible way to support the default OAS serialization for URI parameters + JSON serialization for objects.

## Motivation and Context

Currently we do not honor the OAS specification about parameter serialization for not scalar parameters and instead of that we force JSON serializations for those cases.<br/>
The proposed approach is to provide support for at least the default serializations methods and also keep providing JSON serialization as explained in the OAS spec "Other Serializations Methods" section as workaround for users who want to continue using this behaviour.

## Remaining Work

- [ ] Add support for query and headers parameters
- [ ] Investigate and add tests for edge cases 
- [ ] Inform the user when an unsupported serialization method is defined at the API specification 
- [ ] [Avoid using the JsonParameterValidationStrategy only to use the quote if needed method](https://github.com/mulesoft/apikit-rest-parser-wrapper/pull/358#discussion_r1000952102)

## Related Links

 * [OAS Specification](https://swagger.io/docs/specification/serialization/)
 * [GUS Epic to track this feature](https://gus.lightning.force.com/lightning/r/ADM_Epic__c/a3QEE000000O7Q92AK/view)

### CODE REVIEW

File: interface-impl-amf/src/main/java/org/mule/amf/impl/validation/OASArrayValidationStrategy.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import java.util.stream.Collectors;
+
+import static org.mule.amf.impl.util.AMFUtils.APPLICATION_JSON;
+import static org.mule.amf.impl.util.AMFUtils.needsQuotes;
+import static org.mule.apikit.ParserUtils.quoteValue;
+
+public class OASArrayValidationStrategy implements ParameterValidationStrategy {
+
+  AnyShape arrayShape;
+  JsonParameterValidationStrategy facet;
```
Comment (juanchib) [inline interface-impl-amf/src/main/java/org/mule/amf/impl/validation/OASArrayValidationStrategy.java:30]: I know that using  a **JsonParameterValidationStrategy** only to use the quote if needed method is an overkill but it wasn't easy to extract the logic. Adding this as remaining work.
